django/contrib/gis/db/backends/postgis/base.py

```diff
@@ -27,3 +136,59 @@ def prepare_database(self):
             if bool(cursor.fetchone()):
                 return
             cursor.execute("CREATE EXTENSION IF NOT EXISTS postgis")
+            self.register_geometry_adapters(self.connection)
```

Comment:
Good question, I guess it depends on whether the connection that is used to prepare the database is used for other stuff afterwards that requires the oids to be known. As your tests show it probably doesn't get used in any way requiring that information (currently), but I'd probably keep it since it is only used by `migrate` and as such not exactly performance critical. That said a comment will probably not hurt.